src/pages/docs/creators/javascript/tutorials/rest-api-with-express.md

```diff
@@ -0,0 +1,278 @@
+
+## Prerequisites
+
+This tutorial assumes that you have already installed Yagna and have it running in the background. If you haven't done so yet, please follow the instructions in [this tutorial](/docs/creators/javascript/examples/tools/yagna-installation-for-requestors) before proceeding.
+
+## Setting up the project
+
+First, create a new directory for your project and initialize a new Node.js project in it:
+
+```bash
+mkdir golem-express
+cd golem-express
+npm init -y
+npm install @golem-sdk/golem-js express
+```
+
+## Creating the API
+
+Let's start by creating a new file called `index.mjs` and pasting the following code into it:
+
+```js
+import express from 'express'
+
+const app = express()
+const port = 3000
+
+app.use(express.text())
+
+app.get('/', (req, res) => {
+  res.send('Hello World!')
+})
+
+app.listen(port, () => {
+  console.log(`Example app listening at http://localhost:${port}`)
+})
+```
+
+Start your app by running `node index.mjs` and run `curl http://localhost:3000` in another terminal window. You should see the following output:
+
+```bash
+Hello World!
+```
+
+So far so good! Stop the app by pressing `Ctrl+C` and let's move on to the next step.
+
+## Connecting to the Golem Network
+
+Now let's connect to the Golem Network. First, import the `GolemNetwork` class from `@golem-sdk/golem-js` and create a new instance of it. If you're feeling adventurous, [you can create your own image](/docs/creators/javascript/examples/tools/converting-docker-image-to-golem-format#building-your-docker-image) and install [espeak](https://espeak.sourceforge.net/) on it, but to save time we will use the image `severyn/espeak:latest` provided by one of our community members.
+
+```js
+import { GolemNetwork } from '@golem-sdk/golem-js'
+
+const golem = new GolemNetwork({
+  image: 'severyn/espeak:latest',
+  yagnaOptions: { apiKey: 'try_golem' },
+})
+golem
+  .init()
+  .then(() => {
+    console.log('Connected to the Golem Network!')
+  })
+  .catch((err) => {
+    console.error('Error connecting to the Golem Network:', err)
+    process.exit(1)
+  })
+```
+
+Let's also add a handler for the `SIGINT` signal so that we can close the connection to the Golem Network when the app is stopped.
+
+```js
+process.on('SIGINT', async () => {
+  await golem.close()
+  process.exit(0)
+})
+```
+
+## Creating a retrievable task
+
+Let's add a new endpoint to our API that will take some text from request body and create a new task on the Golem Network. To do that, we will use the `createJob()` method. This method will give us a `Job` object that we can use to get the state of the job and it's results later. On the provider side we will run the `espeak` command to convert the text to speech, save it to a `.wav` file download that file to your local filesystem with the `downloadFile()` method. We will give the file a random name to avoid collisions.
```

Comment:
from **the** request body

Reply:
> On the provider side we will run the `espeak` command to convert the text to speech, save it to a `.wav` file download that file to your local filesystem with the `downloadFile()` method. We will give the file a random name to avoid collisions.

add a **and** to the end of the sentence so it sounds like the following:

"On the provider side we will run the `espeak` command to convert the text to speech, save it to a `.wav` file **and** download that file to your local filesystem with the `downloadFile()` method. We will give the file a random name to avoid collisions."